### PR TITLE
add state=empty into file module

### PR DESCRIPTION
**Pull request by Nindaleth**
Pull request for issue #902
###### Current behaviour:
- Directories: remove all files and subdirectories inside so that the directory contains nothing
- Files: truncate the target file if not empty
- Symlinks/Hardlinks: check which of the previous 2 types the target is, act accordingly on the target

### CODE REVIEW

**Comment by gregdek** [comment]
@Nindaleth This module is maintained by the Ansible core team, so it may take a while longer to get a review; we appreciate your patience. attn: @bcoca @abadger @jimi-c @bennojoy

**Comment by gregdek** [comment]
@Nindaleth thanks for your patience here.

Since we're in the process of pushing Ansible 2.0 out the door, it's unlikely that we'll be able to look at this PR before then. Marking with "next" milestone to indicate this.

**Comment by robynbergeron** [comment]
Thanks @Nindaleth for this PR. Unfortunately, it is not mergeable in its current state due to merge conflicts. Please rebase your PR. When you are done, please comment with text 'ready_for_review' and we will put this PR back into review.

**Comment by Nindaleth** [comment]
ready_for_review

**Comment by Nindaleth** [comment]
I rebased the PR, can you please remove the 'needs_rebase' and add back the 'core_review'?

**Comment by thomasleduc** [comment]
I absolutely need this PR, how can we workaround while waiting for this to be merged ?

**Comment by jhawkesworth** [comment]
@thomasleduc - you can probably copy the whole file into your ansible configuration and set your library in ansible.cfg  - usually somewhere like /etc/ansible/library or /usr/share/library see http://docs.ansible.com/ansible/intro_configuration.html#library

then this version of the file.py will be used instead of the supplied version.

**Comment by thomasleduc** [comment]
Thx @jhawkesworth, but I can't use this solution : 
- I want to continue update ansible.
- I share the ansible project with my team.

I just make a dirty bash command to register an ansible variable waiting for the merge :cry:

**Comment by jhawkesworth** [comment]
You can continue to update ansible and share ansible project with the team.

I do this by keeping all ansible configuration in source control.

When we move to new version of ansible we can just remove custom modules from source control.

Hope this helps.

**Comment by gregdek** [comment]
Thanks @Nindaleth for this PR. This module is maintained by the Ansible core team, so it can take a while for patches to be reviewed. Thanks for your patience.

File: files/file.py [hunk 1]
```diff
@@ -63,9 +63,10 @@
         If C(touch) (new in 1.4), an empty file will be created if the C(path) does not
         exist, while an existing file or directory will receive updated file access and
         modification times (similar to the way `touch` works from the command line).
+        If C(empty) (new in 2.0), directories and files or symlinks will be emptied, but kept.
```
Comment (resmo) [inline files/file.py]: Seems this will make it into 2.1, would you mind updating the version here?

**Comment by Nindaleth** [comment]
@resmo Not a problem, I updated the version in doc.

**Comment by alikins** [comment]
See also https://github.com/ansible/ansible-modules-core/pull/520/ (add a 'create' state). 
Not the same idea, but vaguely similar goals.

**Comment by gregdek** [comment]
Thanks @Nindaleth for this PR. Unfortunately, it is not mergeable in its current state due to merge conflicts. Please rebase your PR. When you are done, please comment with text 'ready_for_review' and we will put this PR back into review.

For help on how to do this cleanly please see http://docs.ansible.com/ansible/community.html#contributing-code-features-or-bugfixes

[This message brought to you by your friendly Ansibull-bot.]

**Comment by gregdek** [comment]
@Nindaleth A friendly reminder: this pull request has been marked as needing your action. If you still believe that this PR applies, and you intend to address the issues with this PR, just let us know in the PR itself and we will keep it open pending your changes. When you do address the issues, please respond with ready_for_review in your comment, so that we can notify the maintainer.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by gregdek** [comment]
@Nindaleth Another friendly reminder: this pull request has been marked as needing your action. If you still believe that this PR applies, and you intend to address the issues with this PR, just let us know in the PR itself and we will keep it open. If you have addressed the issues and believe it's ready for review, please comment with the text "ready_for_review". If we don't hear from you within another 14 days, we will close this pull request.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by gregdek** [comment]
@Nindaleth A friendly reminder: this pull request has been marked as needing your action. If you still believe that this PR applies, and you intend to address the issues with this PR, just let us know in the PR itself and we will keep it open pending your changes. When you do address the issues, please respond with ready_for_review in your comment, so that we can notify the maintainer.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by Nindaleth** [comment]
I'm abandoning this; the code wasn't that good in the first place... someone else, please step forward with a better PR.

**Pull request closed without merging**